### PR TITLE
Lifestyle Budget App 🌂👖👗👘👓

### PR DESCRIPTION
**Pull request by joyapisi**
## Work Done:
Day 1
- Set up the repository and tools.
- Create models according to the given ERD diagram
- Do validations 
- Implement authentication and authorization for login with Devise gem.

Day 2
- Create the home page.
- Create the page to add a category.
- Create the transactions page.
- Create the page to add a transaction.
- Style my app

Day 3 & 4
- Do unit and integration tests.
- Deploy the project, and test for final details.
- Record a video to explain my project.
- Create a good README and PR description.

### CODE REVIEW

File: app/views/landing_page/index.html.erb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+<div class="landing-page">
+  <h1>Lifestyle Budget App</h1>
+  <div class="lower-part">
+    <span class=""><%= link_to "LOG IN", new_user_session_path %></span>
+    <span class=""><%= link_to "SIGN UP", new_user_registration_path %></span>
+  </div>
+</div>
```
Comment (anagudelogu) [inline app/views/landing_page/index.html.erb]: - [ ] Please make sure to follow the design guidelines:

|Template|Your design|
|---|---|
| image | image |

According to the requirements:

image

Please consider working on this! 🤓 🔧

File: app/views/groups/index.html.erb [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+<br/>
+
+<div id="groups">
+  <% @groups.each do |group| %>
+    <%= link_to group_path(group), class: "group-link" do %>
+      <%= render group %>
+    <% end %>
+  <% end %>
+</div>
```
Comment (anagudelogu) [inline app/views/groups/index.html.erb]: - [ ] Also for your "Categories" page:

|Template|Your design|
|---|---|
| image | image |

The button should be fixed to the bottom, each category card should fit the width of the screen and the navigation bar could be improved to match the design! 🤓 🔧

File: app/views/groups/show.html.erb [hunk 3]
```diff
@@ -0,0 +1,9 @@
+<p style="color: green"><%= notice %></p>
+<br/>
+<%= render @group %>
+
+<div>
+  <%= link_to "Back to groups", groups_path, class:"inner-buttons" %>
+
+  <%= button_to "Destroy this group", @group, method: :delete, class:"new-buttons" %>
+</div>
```
Comment (anagudelogu) [inline app/views/groups/show.html.erb]: - [ ] According to the requirements:

image

When I click on a category, I'm unable to see the transactions registered under that category. Please make sure to implement it!

|Template|Your design|
|---|---|
| image | image |

Also, please make sure to include a "Add transaction" button at the bottom of the screen!

Consider making this changes 🤓 🔧

**Review by anagudelogu — CHANGES_REQUESTED** [review]
# STATUS:  CHANGES REQUIRED  ♻️ 🚧 

Hi @joyapisi, 
I'm Andres.


While you have done a great job working on this milestone 💻 , there are some issues that you still need to work on to go to the next project. ♻️ 

You are almost there, keep up the good work! 🚀 



## Highlights
✅ Great use of Git flow. 💯 
✅ Your project is documented professionally. 📜 
✅ Descriptive PR title, summary and commit messages. 💬 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/views/groups/show.html.erb (hunk 3 above)
Comment (VICTOR-CHUKWU) [reply app/views/groups/show.html.erb]: @joyapisi  this is not done yet

File: app/views/groups/index.html.erb [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+  <div id="groups">
+    <% @groups.each do |group| %>
+      <%= link_to group_path(group), class: "group-link" do %>
+        <%= render group %>
+      <% end %>
+    <% end %>
+  </div>
+  
+  <div class="footer-buttons">
+  <%= link_to "View Clothes", group_clothes_path(:group_id), class:"inner-buttons"%>
```
Comment (VICTOR-CHUKWU) [inline app/views/groups/index.html.erb]: - [ ] I really don't know what you mean by view clothes, do you mean view transactions?

File: app/views/groups/show.html.erb [hunk 5]
```diff
@@ -0,0 +1,9 @@
+<p style="color: green"><%= notice %></p>
+<br/>
+<%= render @group %>
+
+<div class="footer-buttons">
+  <%= link_to "Back to groups", groups_path, class:"inner-buttons" %>
+
+  <%= button_to "Destroy this group", @group, method: :delete, class:"new-buttons" %>
+</div>
```
Comment (VICTOR-CHUKWU) [inline app/views/groups/show.html.erb]: - [ ] just as the previous reviewer mentioned, this page should show a list of transactions made for this particular group and also a button to add more transactions.
your current design
Screenshot from 2023-08-12 00-46-55

**Review by VICTOR-CHUKWU — CHANGES_REQUESTED** [review]
Hi @joyapisi , 
## STATUS: CHANGES REQUIRED :recycle: 

Great work on making the changes 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


## Suggested changes
_Check the comments under the review._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

File: app/views/groups/show.html.erb (hunk 5 above)
Comment (codepantha) [inline app/views/groups/show.html.erb]: - [ ] As suggested by the previous reviewer, please ensure that this page shows a list of transactions and also a button to add more transactions at the bottom of the page 🙏

**Review by codepantha — CHANGES_REQUESTED** [review]
# STATUS: CHANGES REQUIRED ♻️ 

## Hi @joyapisi 👋, 

Great work on making the changes requested by the previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.



## Highlights 🎯 
✔️ Descriptive PR
✔️ No linter errors 🟢 
✔️ Good use of GitFlow 📌 

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/views/groups/show.html.erb (hunk 5 above)
Comment (Abenezer-Tilahun) [reply app/views/groups/show.html.erb]: Fixed.

**Review by Abenezer-Tilahun — APPROVED** [review]
Hi @joyapisi  
 
Your project is complete  👍 💯   There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights 👏🏻

✔️ No linter errors.
✔️ You used correct Git-Flow.
✔️ Professional README  
✔️ Good Commit Messages
✔️ Good job for Implemented Suggested changes.

## Optional suggestions 
 
_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

**Pull request merged**